psh/test-touch.py

```diff
@@ -0,0 +1,114 @@
+# Phoenix-RTOS
+#
+# phoenix-rtos-tests
+#
+# psh touch command text
+#
+# Copyright 2022 Phoenix Systems
+# Author: Damian Loewnau
+#
+# This file is part of Phoenix-RTOS.
+#
+# %LICENSE%
+#
+
+import psh.tools.psh as psh
+from psh.tools.common import (CHARS, assert_present, assert_file_created, assert_random_files,
+                              get_rand_strings, create_testdir)
+
+ROOT_TEST_DIR = 'test_touch_dir'
+
+
+def is_hello_present(p):
+    psh.assert_prompt_after_cmd(p, 'ls -d /usr/bin')
+    ret_code = psh.get_exit_code(p)
+
+    if ret_code != 0:
+        return False
+
+    usrbin_content = psh.ls_simple(p, '/usr/bin')
+    return True if 'hello' in usrbin_content else False
+
+
+def assert_timestamp_change(p, touched_files, uptimes, dir=''):
+    files = psh.ls(p, dir)
+    for file in files:
+        if file.name in touched_files:
+            uptime = uptimes[file.name]
+            msg = f"The file's timestamp hasn't been changed after touch: {dir}/{file.name}, timestamp = {file.date}"
+            # uptime has to be got before touch
+            # to prevent failed assertion when typing uptime in 00:59 and touch in 01:00
+            # 1 min margin is applied
+            next_minute = f'{(int(uptime.minute)+1):02d}'
+            if file.date.time == f'{uptime.hour}:{next_minute}':
+                uptime_str = f'{uptime.hour}:{next_minute}'
+            else:
+                uptime_str = f'{uptime.hour}:{uptime.minute}'
+            # when writing test, system date is always set to Jan 01 00:00
+            # so all touched files should have the following timestamp: Jan 01 00:00 + uptime
+            assert file.date == ('Jan', '1', uptime_str), msg
```

Comment:
I've not check changing timestamp for test files created during the test execution intentionally, because in fact there wasn't way to do it before first minute. That's why I've decided to change timestamps of devices/executables/symlinks, where the compilation date is different (excluding Jan 1 0:00)